identify/README.md

```diff
@@ -133,3 +134,14 @@ clients only support initiating requests while some servers (only) support
 responding to requests. To prevent clients from initiating requests to other
 clients, which given them being clients they fail to respond, clients should not
 advertise `foo` in their `protocols` list.
+
+### signedPeerRecord
+
+This is a serialized [SignedEnvelope][envelope-rfc] containing a [PeerRecord][peer-record-rfc],
+signed by the sending node. It contains the same addresses as the `listenAddrs` field, but in a form that lets us share authenticated addrs with other peers.
+
+This field was introduced in a backwards compatible manner (meaning that it is sent along with the `listenAddrs` field), therefore, it is optional and may be omitted by older implementations. If the `signedPeerRecord` is present, implementations MUST use the data contained within it and ignore duplicated fields present in the main identify message
```

Comment:
```suggestion
This field was introduced in a backwards compatible manner (meaning that it is sent along with the `listenAddrs` field), therefore, it is optional and may be omitted by older implementations. If the `signedPeerRecord` is present, implementations SHOULD use the data contained within it and ignore duplicated fields present in the main identify message
```

If a particular implementation is keen on forcing the peer to use addresses from signed peer record instead of `listenAddrs` field it can leave the `listenAddrs` field blank.

Reply:
The point is what to do if the addresses in `listenAddrs` and the signed peer record are different.

Do you merge them or favour one over the other?  If we specify `MUST` it makes the behaviour unequivocal.

Reply:
> On a second reading, I realised that if it's a MUST it would mean that rust-libp2p is technically not following the spec.

Sort of?  It's just following an earlier version of the spec.

I think this is fine, see also #502 which adds a new optional field with a `MUST` in the description - this would also make all current implementations non-compliant.